tests/modules/site-health/audit-autoloaded-options/audit-autoloaded-options-test.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * Tests for audit-autoloaded-options module.
+ *
+ * @package performance-lab
+ */
+class Audit_Autoloaded_Options_Tests extends WP_UnitTestCase {
+
+	const WARNING_AUTOLOADED_SIZE_LIMIT_IN_BYTES = 800000;
+	const AUTOLOADED_OPTION_KEY                  = 'test_set_autoloaded_option';
+
+	/**
+	 * @dataProvider provider_added_test_info_site_health
+	 *
+	 * Tests perflab_aao_add_autoloaded_options_test()
+	 */
+	public function test_perflab_aao_add_autoloaded_options_test( $provider_added_test_info_site_health ) {
+		$this->assertEqualSets( $provider_added_test_info_site_health, perflab_aao_add_autoloaded_options_test( array() ) );
+	}
+
+	/**
+	 * @dataProvider provider_autoloaded_options_less_than_limit
+	 *
+	 * Tests perflab_aao_autoloaded_options_test() when autoloaded options less than warning size.
+	 */
+	public function test_perflab_aao_autoloaded_options_test_no_warning( $provider_autoloaded_options_less_than_limit ) {
+		$this->assertEqualSets(
+			perflab_aao_autoloaded_options_test(),
+			$provider_autoloaded_options_less_than_limit
+		);
+	}
+
+	/**
+	 * @dataProvider provider_autoloaded_options_bigger_than_limit
+	 *
+	 * Tests perflab_aao_autoloaded_options_test() when autoloaded options more than warning size.
+	 */
+	public function test_perflab_aao_autoloaded_options_test_warning( $provider_autoloaded_options_bigger_than_limit ) {
+		self::set_autoloaded_option( self::WARNING_AUTOLOADED_SIZE_LIMIT_IN_BYTES );
+		$this->assertEqualSets(
+			perflab_aao_autoloaded_options_test(),
+			$provider_autoloaded_options_bigger_than_limit
+		);
+	}
+
+	/**
+	 * Tests perflab_aao_autoloaded_options_size()
+	 */
+	public function test_perflab_aao_autoloaded_options_size() {
+		global $wpdb;
+		$autoloaded_options_size = $wpdb->get_var( 'SELECT SUM(LENGTH(option_value)) FROM ' . $wpdb->prefix . 'options WHERE autoload = \'yes\'' );
+		$this->assertEquals( $autoloaded_options_size, perflab_aao_autoloaded_options_size() );
+
+		// Add autoload option.
+		$test_option_string       = 'test';
+		$test_option_string_bytes = mb_strlen( $test_option_string, '8bit' );
+		self::set_autoloaded_option( $test_option_string_bytes );
+		$this->assertSame( $autoloaded_options_size + $test_option_string_bytes, perflab_aao_autoloaded_options_size() );
+	}
+
+	/**
+	 * Sets a test autoloaded option.
+	 *
+	 * @param int $bytes bytes to load in options.
+	 */
+	public static function set_autoloaded_option( $bytes = 800000 ) {
+		$heavy_option_string = self::random_string_generator( $bytes );
+		add_option( self::AUTOLOADED_OPTION_KEY, $heavy_option_string );
+	}
+
+	/**
+	 * Deletes test autoloaded option.
+	 */
+	public static function delete_autoloaded_option() {
+		delete_option( self::AUTOLOADED_OPTION_KEY );
+	}
+
+	/**
+	 * Generate random string with certain $length.
+	 *
+	 * @param int $length Length ( in bytes ) of string to create.
+	 * @return string
+	 */
+	protected static function random_string_generator( $length ) {
```

Comment:
Can you use `wp_generate_password` instead of adding this function?

Reply:
changed for simplicity sake